_posts/2019-07-25-testy-frontendu-okiem-full-stacka.md

```diff
@@ -0,0 +1,136 @@
+---
+layout:    post
+title:     "Testy jednostkowe frontendu okiem programisty full stack"
+published: true
+date:      2019-07-25 10:00:00 +0100
+author:    mmendlik
+tags:
+    - unit test
+    - jasmine
+    - angular
+---
+
+Niełatwo znaleźć wymówkę, żeby nie pisać testów jednostkowych. Obecność frameworków ułatwiających tę czynność w projektach, z którymi stykamy się na co dzień, nie powinna na żadnym chociaż trochę doświadczonym programiście robić wrażenia. Nie trzeba pewnie Cię przekonywać, że **jedne z wielu zalet pisania testów jednostkowych**, to:
+* zmuszenie twórcy do zastanowienia się nad zadaniem sprawdzanego kodu (co może potencjalnie poprawić architekturę aplikacji),
+* ułatwienie wczesnego wyłapywania błędów,
+* ekspozycja przypadków brzegowych,
+* ułatwienie zrozumienia działania kodu osobom, które go nie tworzyły.
+
+W wielu nowoczesnych aplikacjach internetowych, w tym np. we wnioskach Eximee, **duża część logiki znajduje się po stronie klienckiej**, hipokryzją byłoby pominięcie testów w tak istotnym elemencie aplikacji, bo przecież wiadomo, że najszybciej wychodzą na jaw błędy właśnie tam. Niemniej nawet przy ogromnej liczbie narzędzi wspomagających proces pisania testów jednostkowych, programiści mogą mieć problem z **wyznaczeniem właśnie tych jednostek**.
+## Co testować
+Niezależnie od tego jaki framework został użyty w danym projekcie, zawsze możemy z niego wydzielić komponenty. Przeważnie jest to JavaScriptowa klasa z jakimś odniesieniem do szablonu HTML. W praktyce możemy podzielić te komponenty na dwa typy: 
```

Comment:
Zaznaczyłbym, że w tym artykule pokażemy to na przykładzie Angulara

Reply:
ok